src/docbuilder/caching.rs

```diff
@@ -0,0 +1,112 @@
+use crate::utils::copy_dir_all;
+use anyhow::{Context as _, Result};
+use std::{
+    fs, io,
+    path::{Path, PathBuf},
+};
+use tracing::{debug, instrument, warn};
+
+/// move cache folder to target, falling back to copy + delete on error.
+fn move_or_copy<P: AsRef<Path> + std::fmt::Debug, Q: AsRef<Path> + std::fmt::Debug>(
+    source: P,
+    dest: Q,
+) -> io::Result<()> {
+    if let Some(parent) = dest.as_ref().parent() {
+        fs::create_dir_all(parent)?;
+    }
+    if let Err(err) = fs::rename(&source, &dest) {
```

Comment:
Should this be `symlink` instead of `move`? 

On a linux machine, or our server, are the directories on the same filesystem?

Reply:
"the directories" doesn't make a lot of sense to me because we control where these are created, but everything under `DOCSRS_PREFIX` is under the same filesystem, yes.

I think it's ok to use `fs::rename` here and panic if it fails; silently taking orders of magnitude longer than expected seems not great since the whole point is to help with performance.

Reply:
I udpated the code, now: 
- rustwide workspace & prefix are on the same filesystem in docker-compose 
- we always `rename`